tests/integration/test_profiler.py

```diff
@@ -650,10 +661,105 @@ def test_profile_tables_with_different_opts(spark, ws, make_schema, make_random)
     input_df.write.format("delta").saveAsTable(table2_name)
 
     profiler = DQProfiler(ws)
-    table_opts = {
-        table1_name: {"remove_outliers": False, "max_null_ratio": 0.5, "sample_fraction": 1.0, "trim_strings": False},
-        table2_name: {"remove_outliers": False, "sample_fraction": 1.0},
+    table_opts = [
+        {
+            "table": table1_name,
+            "options": {
+                "remove_outliers": False,
+                "max_null_ratio": 0.5,
+                "sample_fraction": 1.0,
+                "trim_strings": False,
+            },
+        },
+        {
+            "table": table2_name,
+            "options": {
+                "remove_outliers": False,
+                "sample_fraction": 1.0,
+            },
+        },
+    ]
+    profiles = profiler.profile_tables(tables=[table1_name, table2_name], options=table_opts)
+    expected_rules = {
+        table1_name: [
+            DQProfile(
+                name="is_not_null",
+                column="category",
+                description="Column category has 20.0% of null values (allowed 50.0%)",
+                parameters=None,
+            ),
+            DQProfile(name="is_not_null", column="value", description=None, parameters=None),
+            DQProfile(
+                name="min_max",
+                column="value",
+                description="Real min/max values were used",
+                parameters={"min": 1, "max": 100},
+            ),
+        ],
+        table2_name: [
+            DQProfile(
+                name="is_not_null_or_empty", column="category", description=None, parameters={"trim_strings": True}
+            ),
+            DQProfile(name="is_not_null", column="value", description=None, parameters=None),
+            DQProfile(
+                name="min_max",
+                column="value",
+                description="Real min/max values were used",
+                parameters={"min": 1, "max": 100},
+            ),
+        ],
     }
+
+    for table_name, (stats, rules) in profiles.items():
+        assert len(stats.keys()) > 0, f"Stats did not match expected for {table_name}"
+        assert rules == expected_rules[table_name], f"Rules did not match expected for {table_name}"
+
+
+def test_profile_tables_with_partial_opts_match(spark, ws, make_schema, make_random):
```

Comment:
Test for cases where the options is empty or no matching pattern is found should be added